server/src/intTest/java/org/tctalent/server/repository/db/integrationhelp/DBContainer.java

```diff
@@ -0,0 +1,190 @@
+ * loaded. It must be initialized statically and run beforeAll to ensure the Flyway scripts execute
+ * against it when Spring launches. This setup occurs before Spring takes over, ensuring the
+ * database is created and users are set up correctly so the dump loads as expected. ##
+ * Requirements: - Refer to the <a
+ * href="./resources/testcontainers.properties">testcontainers.properties</a> file. - Set the system
+ * environment variable *TCTALENT_DB_HOME* to specify the location of the dump file. - Ensure
+ * `testcontainers.properties` is in the classpath with the required variables set.
+ * <p>
+ * The first parameter is the name of the dump file (must start with a slash). The second parameter
+ * is the required mount point in the Docker image, with a default value provided. Configuration
+ * functions are separated, so the container setup is streamlined and focused solely on the
+ * container configuration.
+ */
+@Slf4j
+public class DBContainer {
+
+  private static final String DB_NAME = "tctalent";
+  private static final String DB_USER = "tctalent";
+  private static final String DB_PWD = "tctalent";
+  public static final String DEFAULT_DUMP = "/integration-test-dump.sql";
+  public static final String TEST_CONTAINERS_DUMP = "testcontainers.dump.location";
+  public static final String TEST_CONTAINERS_MOUNT = "testcontainers.container.mount";
+
+  @Container
+  public static final PostgreSQLContainer<?> db = new PostgreSQLContainer<>(pgContainerImage())
+      .withDatabaseName(DB_NAME)
+      .withUsername(DB_USER)
+      .withPassword(DB_PWD)
+      .withReuse(true);
+
+  static {
+    log.info("Creating the postgres container.");
+    log.info("Dump file path: " + dumpFilePath());
+    log.info("Container mount path: " + containerMountPath());
+  }
+
+  /**
+   * Starts the database container and initializes the database.
+   */
+  public static void startDBContainer() {
+    db.start();
+    createDbObjects();
+    loadDb();
+    importDumpFileToDatabase();
+  }
+
+  /**
+   * Imports the database dump file to the database container.
+   */
+  private static void importDumpFileToDatabase() {
+    log.info("Importing the database dump.");
+    try {
+      db.execInContainer(psqlCommand());
+    } catch (IOException | InterruptedException e) {
+      log.error("Could not import the database dump file.{}", e.getMessage());
+    }
+    log.info("Done importing the database dump. Connection is: {}", db.getJdbcUrl());
+    log.info("Ready to use: {}", db.isRunning());
+  }
+
+  /**
+   * Creates database objects and users in the database container.
+   */
+  private static void createDbObjects() {
+    log.info("Creating database objects");
+    try {
+      db.execInContainer(createDbCommands());
+      db.execInContainer(createUserCommands());
+      db.execInContainer(superUserCommand());
+    } catch (InterruptedException | IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

**Remember**: Do this only for InterruptedException. Doing this for other types of exceptions may give rise to subtle bugs.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)